src/dotnet-svcutil/lib/src/Metadata/ServiceDescriptor.cs

```diff
@@ -93,7 +97,53 @@ public async Task ImportMetadataAsync(Action<WsdlImporter> onWsdlImporterCreated
                 return;
             }
 
-            await this.metadataDocumentLoader.LoadAsync(cancellationToken).ConfigureAwait(false);
+            //if it's net.pipe url
+            if (MetadataUrl != null && MetadataUrl.Scheme.Equals("net.pipe"))
+            {
+                string tfn;                
+                if(OperationalCtx == OperationalContext.Infrastructure)
+                {
+                    tfn = "net472";
+                }
+                else
+                {
+                    tfn = "net6.0";
```

Comment:
what about net7.0 and above?

Reply:
The tfn value depends on target framework of NamedPipeMetadataImporter.csproj which we used to get metadata for namedpipe binding service, that project multi-targets to net6.0 and net472, so the value here is fixed.

Reply:
Now you mention that this is fixed for both net6.0 and net472, I am not sure that I understand why this couldn't work on net462.

Reply:
net6.0 is the same as the current also the lowest .net version of NetNamedPipe package, net472 is what wcf connected service tool project targets to.

NamedPipeMetadataImporter.csproj targets to net6.0 and net472 so the tfn value here is net6.0 and net472.

The value of variable "tfn" is part of the directory path to locate the internal assets (NamedPipeMetadataImporter.dll) used by dotnet-svcutil-lib to generate proxy.

Hope this clarifies the confusion.

Reply:
Let's use net462 for .NET Framework.